### PR TITLE
Fix aggregate functions in dibi::fetchSingle

### PR DESCRIPTION
**Pull request by jakubvojacek**

### CODE REVIEW

**Comment by staabm** [comment]
I just run the test (without the fix) locally, to find out that this PR is trying to fix this problem:
```
1) Error
The data provider specified for staabm\PHPStanDba\Tests\DbaInferenceTest::testFileAsserts is invalid.
PHPStan\ShouldNotHappenException: Cannot create PHPStan\Type\UnionType with: null, int<-32768, 32767>|null
phar:///Users/staabm/workspace/phpstan-dba/vendor/phpstan/phpstan/phpstan.phar/src/Type/UnionType.php:53
phar:///Users/staabm/workspace/phpstan-dba/vendor/phpstan/phpstan/phpstan.phar/src/Type/UnionType.php:67
/Users/staabm/workspace/phpstan-dba/src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:125
/Users/staabm/workspace/phpstan-dba/src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:92
/Users/staabm/workspace/phpstan-dba/src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:68
/Users/staabm/workspace/phpstan-dba/src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:55
```

would be great you could describe the reasoning behind your changes in the PR description next time.

File: src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -118,6 +118,10 @@ private function reduceResultType(MethodReflection $methodReflection, Type $resu
         } elseif ('fetchPairs' === $methodName && $resultType instanceof ConstantArrayType && 2 === \count($resultType->getValueTypes())) {
             return new ArrayType($resultType->getValueTypes()[0], $resultType->getValueTypes()[1]);
         } elseif ('fetchSingle' === $methodName && $resultType instanceof ConstantArrayType && 1 === \count($resultType->getValueTypes())) {
+            if ($resultType->getValueTypes()[0]->accepts(new NullType(), true)->yes()) {
+                return $resultType->getValueTypes()[0];
+            }
+
             return new UnionType([new NullType(), $resultType->getValueTypes()[0]]);
```
Comment (staabm) [inline src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:125]: I guess the newly added lines are unnecessary and this should work with `TypeCombinator::addNull()` instead

Comment (jakubvojacek) [reply src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php:125]: yes, I wasnt aware of such method. That solves it too. Will push changes shortly

Comment (jakubvojacek) [reply src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php]: hopefully fixed

**Comment by staabm** [comment]
Thank you!

**Pull request merged**